### PR TITLE
Add example for using Client cert auth without roles

### PR DESCRIPTION
**Pull request by PrarthonaPaul**

### CODE REVIEW

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 1]
```diff
@@ -0,0 +1,252 @@
+---
+layout: post
+title: 'Using Certificate Based Authentication with Mutual TLS without Roles'
+date: 2023-06-06
+tags: cert-Generation server-Configuration deploying-App
+synopsis: How to deploy a simple web app on eildfly that uses mutual TLS connection and CLIENT_CERT to authenticate the client.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/eildfly/WildFly :)

Comment (fjuma) [reply _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Maybe we could reword this to something like:

How to secure a web application deployed to WildFly with mutual TLS and the CLIENT_CERT HTTP authentication mechanism.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 2]
```diff
@@ -0,0 +1,252 @@
+---
+layout: post
+title: 'Using Certificate Based Authentication with Mutual TLS without Roles'
+date: 2023-06-06
+tags: cert-Generation server-Configuration deploying-App
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Existing tags can be seen here on the right hand side: https://wildfly-security.github.io/wildfly-elytron/blog/

Maybe we could use the following tags here instead:

client-cert tls authentication

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+synopsis: How to deploy a simple web app on eildfly that uses mutual TLS connection and CLIENT_CERT to authenticate the client. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Perfect!

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+Mutual-TLS with client certificates is essential
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: I don't think the - is needed, so just Mutual TLS

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+tags: cert-Generation server-Configuration deploying-App
+synopsis: How to deploy a simple web app on eildfly that uses mutual TLS connection and CLIENT_CERT to authenticate the client. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+Mutual-TLS with client certificates is essential 
+for secure and authenticated communication between
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/authenticated/encrypted

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+Mutual-TLS with client certificates is essential 
+for secure and authenticated communication between 
+clients and servers in various applications and systems. 
+It involved a handshake process during which both the client
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/involved/involves

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+:toc: macro
+:toc-title:
+
+Mutual-TLS with client certificates is essential 
+for secure and authenticated communication between 
+clients and servers in various applications and systems. 
+It involved a handshake process during which both the client 
+and the server must present their certificate to each other 
+in order to establish a secure connection. This ensures that 
+both parties are authenticated before establushing the connection.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/establushing/establishing

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+Mutual-TLS with client certificates is essential 
+for secure and authenticated communication between 
+clients and servers in various applications and systems. 
+It involved a handshake process during which both the client 
+and the server must present their certificate to each other 
+in order to establish a secure connection. This ensures that 
+both parties are authenticated before establushing the connection.  
+
+This example will walk you through how to secure a web application deployed to WildFly with mutual TLS and the CLIENT_CERT HTTP authentication mechanism. We will be doing so using CLIENT_CERT 
+HTTP authentication.In this example we will be using certificates
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: "We will be doing so using CLIENT_CERT HTTP authentication." can probably be removed since it's mentioned in the first sentence.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+clients and servers in various applications and systems. 
+It involved a handshake process during which both the client 
+and the server must present their certificate to each other 
+in order to establish a secure connection. This ensures that 
+both parties are authenticated before establushing the connection.  
+
+This example will walk you through how to secure a web application deployed to WildFly with mutual TLS and the CLIENT_CERT HTTP authentication mechanism. We will be doing so using CLIENT_CERT 
+HTTP authentication.In this example we will be using certificates 
+signed by an example CA. We will be making use of the 
+`Dynamic-certificates` repository.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: `dynamic-certificates` project

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+
+Please navigate to the elytron server home directory and enter the following command.
+
+```
+
+$SERVER_HOME/bin/jboss-cli.sh --connect 
+
+```
+
+This will connect you to the server, after which you may proceed to configuring the server to establish the connection.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: "to establish the connection" could be removed

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+```
+
+$SERVER_HOME/bin/jboss-cli.sh --connect 
+
+```
+
+This will connect you to the server, after which you may proceed to configuring the server to establish the connection. 
+
+First we need to create two keystores inside the elytron subsystem. This will tell elytron where to go inder to find the 
+keystores. The command for this is:
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Just a minor thing, for brevity, "This will tell elytron where to go inder to find the keystores. The command for this is: " could be removed and the first sentence could just be:

First we need to create two keystores inside the elytron subsystem:

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+First we need to create two keystores inside the elytron subsystem. This will tell elytron where to go inder to find the 
+keystores. The command for this is: 
+
+```
+/subsystem=elytron/key-store=twoWayKS:add(path=/path/to/server.keystore, relative-to=jboss.server.config.dir, credential-reference={clear-text=keystorepass}, type=JKS)
+/subsystem=elytron/key-store=twoWayTS:add(path=/path/to/server.truststore,relative-to=jboss.server.config.dir, credential-reference={clear-text=truststorepass}, type=JKS)
+```
+
+The `relative-to` option specifies that the path that you have entered is relative to the `/standalone/configuration` folder 
+inside your wildfly repository.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/wildfly repository/WILDFLY_HOME directory

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+
+```
+/subsystem=elytron/key-store=twoWayKS:add(path=/path/to/server.keystore, relative-to=jboss.server.config.dir, credential-reference={clear-text=keystorepass}, type=JKS)
+/subsystem=elytron/key-store=twoWayTS:add(path=/path/to/server.truststore,relative-to=jboss.server.config.dir, credential-reference={clear-text=truststorepass}, type=JKS)
+```
+
+The `relative-to` option specifies that the path that you have entered is relative to the `/standalone/configuration` folder 
+inside your wildfly repository. 
+
+Next, we need to create a key-manager and a trust manager to point to `twoWayKS` and `twoWayTS`, the previously
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: `key-manager` and `trust-manager`

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+
+Next, we need to create a key-manager and a trust manager to point to `twoWayKS` and `twoWayTS`, the previously 
+created keystores using the following commands: 
+```
+/subsystem=elytron/key-manager=twoWayKM:add(key-store=twoWayKS,credential-reference={clear-text=keystorepass})
+
+/subsystem=elytron/trust-manager=twoWayTM:add(key-store=twoWayTS)
+```
+
+We also need to have an identity realm, which would help up map our user to one that exists inside the file system.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/up/us

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+We also need to have an identity realm, which would help up map our user to one that exists inside the file system. 
+For this step, let us create a filesystem realm called `idsRealm`. We will enter into this realm our identity with 
+name `client-cert-user` which we will map the certificate into. 
+
+```
+/subsystem=elytron/filesystem-realm=idsRealm:add(path=ids, relative-to=jboss.server.config.dir)
+/subsystem=elytron/filesystem-realm=idsRealm:add-identity(identity=client-cert-user)
+```
+
+Next, we need to configure our server-ssl-context. This is a collection of ciphers, TLS settings and certificates, like the
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: `server-ssl-context`

Comment (fjuma) [reply _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: I think we can remove " This is a collection of ciphers, TLS settings and certificates, like the ones we will be using to authenticate our server and the client."

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+For this step, let us create a filesystem realm called `idsRealm`. We will enter into this realm our identity with 
+name `client-cert-user` which we will map the certificate into. 
+
+```
+/subsystem=elytron/filesystem-realm=idsRealm:add(path=ids, relative-to=jboss.server.config.dir)
+/subsystem=elytron/filesystem-realm=idsRealm:add-identity(identity=client-cert-user)
+```
+
+Next, we need to configure our server-ssl-context. This is a collection of ciphers, TLS settings and certificates, like the 
+ones we will be using to authenticate our server and the client. Since the `applicationSSC` server-SSL-context already exists,
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/exists,/exists in the default configuration,

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+ones we will be using to authenticate our server and the client. Since the `applicationSSC` server-SSL-context already exists, 
+we will not be adding one. Instead, we will be editing its properties using the `write-attribute` command: 
+
+```
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=key-manager, value=twoWayKM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=trust-manager, value=twoWayTM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=need-client-auth, value=true)
+```
+
+Now, when we are establishing the connection, elytron will know where to go to look for the keystore and the truststore.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Now, when WildFly receives an HTTPS request, elytron will make use of this SSL context when handling the request.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+we will not be adding one. Instead, we will be editing its properties using the `write-attribute` command: 
+
+```
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=key-manager, value=twoWayKM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=trust-manager, value=twoWayTM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=need-client-auth, value=true)
+```
+
+Now, when we are establishing the connection, elytron will know where to go to look for the keystore and the truststore. 
+Setting `want-client-auth` to true indicates that a client needs to be authenticated in order to establish a connection.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: `need-client-auth`

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 19]
```diff
@@ -0,0 +1,252 @@
...
+```
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=key-manager, value=twoWayKM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=trust-manager, value=twoWayTM)
+/subsystem=elytron/server-ssl-context=applicationSSC:write-attribute(name=need-client-auth, value=true)
+```
+
+Now, when we are establishing the connection, elytron will know where to go to look for the keystore and the truststore. 
+Setting `want-client-auth` to true indicates that a client needs to be authenticated in order to establish a connection. 
+
+Next, we will need to configure a decoder that will be able to decode the CA Certification. We will be using
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: This could be reworded to something like:

Next, we will configure a principal transformer that will map an X.509 certificate to a generic user name. To do this, we will make use of a `chained-principal-transformer` that consists of an `x500-attribute-principal-decoder` and a `regex-principal-transformer`.

The `x500-attribute-principal-decoder` is used to obtain the first segment from the CN value from an X.509 certificate.

The `regex-principal-transformer` checks if the CN value matches a specific pattern and if so maps the principal to `client-cert-user`.

`client-cert-user` is a user without any roles.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 20]
```diff
@@ -0,0 +1,252 @@
...
+a regex-principal-transformer chained with the x500 attribute principal decoder to map usernames with the given pattern 
+into our client-cert-user. 
+
+```
+/subsystem=elytron/x500-attribute-principal-decoder=cnDecoder:add(attribute-name=CN, maximum-segments=1)
+/subsystem=elytron/regex-principal-transformer=myRegexTransformer:add(pattern="[a-zA-Z]+\\.[a-zA-Z]+\\.[0-9]+", replacement="client-cert-user")
+/subsystem=elytron/chained-principal-transformer=myChainedTransformer:add(principal-transformers=[cnDecoder, myRegexTransformer])
+```
+
+Now we will use a security realm to use the filesystem realm fom before:
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Maybe this could be reworded to something like:

Next, we'll create a security domain that references the filesystem realm that was previously created.

You could also move those commands that create the filesystem realm here instead.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 21]
```diff
@@ -0,0 +1,252 @@
...
+/subsystem=elytron/regex-principal-transformer=myRegexTransformer:add(pattern="[a-zA-Z]+\\.[a-zA-Z]+\\.[0-9]+", replacement="client-cert-user")
+/subsystem=elytron/chained-principal-transformer=myChainedTransformer:add(principal-transformers=[cnDecoder, myRegexTransformer])
+```
+
+Now we will use a security realm to use the filesystem realm fom before: 
+```
+/subsystem=elytron/security-domain=clientCertDomain:add(realms=[{realm=idsRealm}], default-realm=idsRealm, pre-realm-principal-transformer=myChainedTransformer, permission-mapper=default-permission-mapper)
+```
+
+Afterwards, use the following commands to configure the CLIENT_CERT HTTP mechanism on elytron:
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: "on elytron" can be removed

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 22]
```diff
@@ -0,0 +1,252 @@
...
+/subsystem=elytron/configurable-http-server-mechanism-factory=configuredCert:add(http-server-mechanism-factory=global, properties={org.wildfly.security.http.skip-certificate-verification=true})
+/subsystem=elytron/http-authentication-factory=clientCertAuth:add(http-server-mechanism-factory=configuredCert, security-domain=clientCertDomain , mechanism-configurations=[{mechanism-name=CLIENT_CERT}])
+```
+
+And finally, we need to configure the undertow subsystem to be able to establish a connection. Please run the following commands
+in a batch as specified below to avoid unneccessary warnings: 
+```
+batch
+#configure the undertow subsystem	
+/subsystem=undertow/server=default-server/https-listener=https:write-attribute(name=ssl-context, value=applicationSSC)
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: This is the default configuration so this line isn't needed.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 23]
```diff
@@ -0,0 +1,252 @@
...
+/subsystem=elytron/security-domain=clientCertDomain:add(realms=[{realm=idsRealm}], default-realm=idsRealm, pre-realm-principal-transformer=myChainedTransformer, permission-mapper=default-permission-mapper)
+```
+
+Afterwards, use the following commands to configure the CLIENT_CERT HTTP mechanism on elytron:
+```
+/subsystem=elytron/configurable-http-server-mechanism-factory=configuredCert:add(http-server-mechanism-factory=global, properties={org.wildfly.security.http.skip-certificate-verification=true})
+/subsystem=elytron/http-authentication-factory=clientCertAuth:add(http-server-mechanism-factory=configuredCert, security-domain=clientCertDomain , mechanism-configurations=[{mechanism-name=CLIENT_CERT}])
+```
+
+And finally, we need to configure the undertow subsystem to be able to establish a connection. Please run the following commands
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/to be able to establish a connection/to make use of our `http-authentication-factory`

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 24]
```diff
@@ -0,0 +1,252 @@
...
+batch
+#configure the undertow subsystem	
+/subsystem=undertow/server=default-server/https-listener=https:write-attribute(name=ssl-context, value=applicationSSC)
+/subsystem=undertow/application-security-domain=other:undefine-attribute(name=security-domain)
+/subsystem=undertow/application-security-domain=other:write-attribute(name=http-authentication-factory,value=clientCertAuth)
+/subsystem=undertow/application-security-domain=other:write-attribute(name=override-deployment-config,value=true)
+run-batch
+```
+
+And to apply all these changes, please type in `reload` into the same terminal. Now that we have established a connection,
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/Now that we have established a connection,/Now that we have configured the server, we are ready to deploy and test our application.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 25]
```diff
@@ -0,0 +1,252 @@
...
+Note that since the server's certificate won't be trusted by your browser, you'll need to manually confirm that
+this certificate is trusted or configure your browser to trust it.
+
+Select the certificate for Bob.Smith.123456 and then click on
+“Access Secured Servlet”. This will authenticate the client and will recognize that Bob accessed the web application and print "Login Successful". 
+
+If you used a certificate where the username does not match our regex pattern, then you will see a "Login Unsuccessful" message, or a `forbidden` 
+error. This is because the server is not set up to map user identity that does not match the regex pattern to the `client-cert-user` identity. 
+
+If you are using the `Bob.Smith.123456` identity, and getting a "Login Unsuccessful" message, please go through the instructions carefully and try again.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: I think this line can be removed.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 26]
```diff
@@ -0,0 +1,252 @@
...
+
+```
+
+$SERVER_HOME/bin/jboss-cli.sh --connect --file=$PATH_TO_ELYTRON_EXAMPLES/client-cert-auth-without-roles/restore-elytron-configuration.cli
+
+```
+
+Alteratively, you can keep a backup of your `standalone.sh` file and replace it with the backup once you want to go back. 
+
+This example has shown to secure a web application deployed to WildFly using the CLIENT_CERT HTTP authentication
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/shown/shown how

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 27]
```diff
@@ -0,0 +1,252 @@
...
+mechanism with two-way SSL with authorization. It has also demonstrated that individual client certificates do not need
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: "with authorization" can be removed since we didn't use any roles

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 28]
```diff
@@ -0,0 +1,241 @@
...
+for secure and encrypted communication between 
+clients and servers in various applications and systems. 
+It involves a handshake process during which both the client 
+and the server must present their certificate to each other 
+in order to establish a secure connection. This ensures that 
+both parties are authenticated before establishing the connection.  
+
+This example will walk you through how to secure a web application deployed to WildFly with mutual TLS and the CLIENT_CERT HTTP authentication mechanism. In this example we will be using certificates 
+signed by an example CA. We will be making use of the 
+`Dynamic-certificates` project.
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: `dynamic-certificates`

**Comment by fjuma** [comment]
@PrarthonaPaul Are you able to add the link to the built Elytron website from your fork so we can preview the post?

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 29]
```diff
@@ -0,0 +1,233 @@
+---
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Please change the file name to today's date (so the right date appears when we merge it).

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 30]
```diff
@@ -0,0 +1,233 @@
...
+layout: post
+title: 'Using Certificate Based Authentication with Mutual TLS without Roles'
+date: 2023-06-06
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Please update to today's date.

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 31]
```diff
@@ -0,0 +1,233 @@
...
+
+cp $PATH_TO_ELYTRON_EXAMPLES/dynamic-certificates/target/server.* $WILDFLY_HOME/standalone/configuration
+
+cp $PATH_TO_ELYTRON_EXAMPLES/dynamic-certificates/target/client* $WILDFLY_HOME/standalone/configuration
+
+```
+
+== Server Configuration
+
+The following set of instructions are going to be used to configure the Wildfly server in order to
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/Wildfly/WildFly

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc [hunk 32]
```diff
@@ -0,0 +1,233 @@
...
+```
+
+Afterwards, use the following commands to configure the CLIENT_CERT HTTP mechanism:
+```
+/subsystem=elytron/configurable-http-server-mechanism-factory=configuredCert:add(http-server-mechanism-factory=global, properties={org.wildfly.security.http.skip-certificate-verification=true})
+/subsystem=elytron/http-authentication-factory=clientCertAuth:add(http-server-mechanism-factory=configuredCert, security-domain=clientCertDomain , mechanism-configurations=[{mechanism-name=CLIENT_CERT}])
+```
+
+And finally, we need to configure the undertow subsystem to make use of our `http-authentication-factory`. Please run the following commands
+in a batch as specified below to avoid unneccessary warnings:
```
Comment (fjuma) [inline _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: s/unneccessary/unnecessary

**Comment by PrarthonaPaul** [comment]
> @PrarthonaPaul Are you able to add the link to the built Elytron website from your fork so we can preview the post?

For some reason https://prarthonapaul.github.io/wildfly-elytron/blog/ gives a 404 error. My guess is that the build fails each time. 
I was only able to access it using http://localhost:4000/wildfly-elytron/

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc (hunk 31 above)
Comment (PrarthonaPaul) [reply _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: Fixed!

File: _posts/2023-06-06-Client-cert-auth-without-roles.adoc (hunk 32 above)
Comment (PrarthonaPaul) [reply _posts/2023-06-06-Client-cert-auth-without-roles.adoc]: fixed!

**Comment by fjuma** [comment]
> > @PrarthonaPaul Are you able to add the link to the built Elytron website from your fork so we can preview the post?
> 
> For some reason https://prarthonapaul.github.io/wildfly-elytron/blog/ gives a 404 error. My guess is that the build fails each time. I was only able to access it using http://localhost:4000/wildfly-elytron/

@cam-rod Are you able to check if you can successfully push this to your fork?

**Comment by jessicarod7** [comment]
Resolved build issues in #1920

File: _posts/2023-06-15-Client-cert-auth-without-roles.adoc [hunk 33]
```diff
@@ -0,0 +1,241 @@
...
+
+- Click the *View Certificates* button.
+
+- A new window will open. Select the *Your Certificates* tab and click the *Import* button.
+
+- Select the *client1.keystore.pkcs12* file. You will be prompted to enter the password: `keystorepass`.
+
+- The certificate is now installed in the Mozilla Firefox browser.
+
+You can follow the same instructions to import `client2.keystore.pkcs12` into your browser. Now try accessing the application using https://localhost:8443/simple-webapp.
```
Comment (fjuma) [inline _posts/2023-06-15-Client-cert-auth-without-roles.adoc]: "Now try accessing..." could be moved to a new line and the section could be called `== Access the Application`.

**Comment by fjuma** [comment]
Placing this on hold until @PrarthonaPaul is back. When back, please:

* Update the file name with the current date.
* Update the date in the file with the current date.
* Follow the steps from @cam-rod's PR to see if that allows the push to your gh-pages branch to succeed: https://github.com/wildfly-security/wildfly-elytron/pull/1921
* Move the text about accessing the app to a new section called `Access the Application` instead of having that with the import section.

**Comment by PrarthonaPaul** [comment]
> Placing this on hold until @PrarthonaPaul is back. When back, please:
> 
> * Update the file name with the current date.
> * Update the date in the file with the current date.
> * Follow the steps from @cam-rod's PR to see if that allows the push to your gh-pages branch to succeed: [Update GitHub token instructions to enable Actions permission #1921](https://github.com/wildfly-security/wildfly-elytron/pull/1921)
> * Move the text about accessing the app to a new section called `Access the Application` instead of having that with the import section.

All changes applied.

**Comment by PrarthonaPaul** [comment]
Blog posted: https://prarthonapaul.github.io/wildfly-elytron/blog/Client-cert-auth-without-roles/

**Comment by fjuma** [comment]
> Blog posted: https://prarthonapaul.github.io/wildfly-elytron/blog/Client-cert-auth-without-roles/

Thanks! From there I see:

[Import the Certificate into Your Browser](https://prarthonapaul.github.io/wildfly-elytron/blog/Client-cert-auth-without-roles/#import-the-certificate-into-your-browser)
[Import the Certificate into Google Chrome](https://prarthonapaul.github.io/wildfly-elytron/blog/Client-cert-auth-without-roles/#import-the-certificate-into-google-chrome)
[Import the Certificate into Mozilla Firefox](https://prarthonapaul.github.io/wildfly-elytron/blog/Client-cert-auth-without-roles/#import-the-certificate-into-mozilla-firefox)

I think those last two headings should be subheadings, right?

File: _posts/2023-06-26-Client-cert-auth-without-roles.adoc [hunk 34]
```diff
@@ -0,0 +1,245 @@
...
+mvn clean install wildfly:deploy
+```
+
+*Note:* There is also a repository called `elytron-examples/simple-webapp`. But we will not be using that. Instead, we will be deploying a slightly modified version of that web application located inside this repository.
+
+== Import the Certificate into Your Browser
+
+Before you access the application, you must import the _client1.keystore.pkcs12_, which holds the client certificate, into your browser.
+
+== Import the Certificate into Google Chrome
```
Comment (fjuma) [inline _posts/2023-06-26-Client-cert-auth-without-roles.adoc]: ===

File: _posts/2023-06-26-Client-cert-auth-without-roles.adoc [hunk 35]
```diff
@@ -0,0 +1,245 @@
...
+
+- Scroll down to the *Advanced* section and on the *Manage certificates* screen, select the *Your Certificates* tab and click on the *Import* button.
+
+- Select the *client1.keystore.pkcs12* file. You will be prompted to enter the password: `keystorepass`.
+
+- The client certificate is now installed in the Google Chrome browser.
+
+You can follow the same instructions to import `client2.keystore.pkcs12` into your browser. 
+
+== Import the Certificate into Mozilla Firefox
```
Comment (fjuma) [inline _posts/2023-06-26-Client-cert-auth-without-roles.adoc]: ===

File: _posts/2023-06-26-Client-cert-auth-without-roles.adoc (hunk 34 above)
Comment (PrarthonaPaul) [reply _posts/2023-06-26-Client-cert-auth-without-roles.adoc]: fixed

File: _posts/2023-06-26-Client-cert-auth-without-roles.adoc (hunk 35 above)
Comment (PrarthonaPaul) [reply _posts/2023-06-26-Client-cert-auth-without-roles.adoc]: fixed

**Review by fjuma — APPROVED** [review]
Thanks @PrarthonaPaul!
